File: apps/api/app/Constants/SocketEvents.php
<?php

namespace App\Constants;

/**
 * Must stay in sync with packages/shared-types (frontend copies to Dart).
 */
class SocketEvents
{
    // Driver → server
    public const DRIVER_LOCATION_UPDATE = 'driver:location:update';
    public const DRIVER_STATUS_UPDATE = 'driver:status:update';
    public const RIDE_ACCEPT = 'ride:accept';
    public const RIDE_DECLINE = 'ride:decline';
    public const RIDE_ARRIVED = 'ride:arrived';
    public const RIDE_CASH_CONFIRM = 'ride:cash:confirm';
    public const RIDE_COMPLETE = 'ride:complete';

    // Passenger → server
    public const RIDE_CANCEL = 'ride:cancel';

    // Server → driver
    public const SERVER_RIDE_DISPATCHED = 'server:ride:dispatched';
    public const SERVER_RIDE_CANCELLED = 'server:ride:cancelled';
    public const SERVER_RIDE_TIMEOUT = 'server:ride:timeout';

    // Server → passenger
    public const SERVER_RIDE_ACCEPTED = 'server:ride:accepted';
    public const SERVER_DRIVER_LOCATION = 'server:driver:location';
    public const SERVER_DRIVER_ARRIVED = 'server:driver:arrived';
    public const SERVER_RIDE_STARTED = 'server:ride:started';
    public const SERVER_RIDE_COMPLETED = 'server:ride:completed';
    public const SERVER_DRIVER_CANCELLED = 'server:driver:cancelled';

    // Admin
    public const ADMIN_SOS_ALERT = 'admin:sos:alert';
    public const ADMIN_DASHBOARD = 'admin:dashboard';
}

File: apps/driver-app/lib/constants/socket_events.dart
/// Mirror of apps/api/app/Constants/SocketEvents.php. Keep in sync when
/// the server gains a new event; the build will break if either side
/// drifts, which is intentional.
class SocketEvents {
  SocketEvents._();

  // Driver → server
  static const String driverLocationUpdate = 'driver:location:update';
  static const String driverStatusUpdate = 'driver:status:update';
  static const String rideAccept = 'ride:accept';
  static const String rideDecline = 'ride:decline';
  static const String rideArrived = 'ride:arrived';
  static const String rideCashConfirm = 'ride:cash:confirm';
  static const String rideComplete = 'ride:complete';

  // Passenger → server (driver doesn't emit these; listed for parity)
  static const String rideCancel = 'ride:cancel';

  // Server → driver
  static const String serverRideDispatched = 'server:ride:dispatched';
  static const String serverRideCancelled = 'server:ride:cancelled';
  static const String serverRideTimeout = 'server:ride:timeout';

  // Server → passenger (driver doesn't receive these; listed for parity)
  static const String serverRideAccepted = 'server:ride:accepted';
  static const String serverDriverLocation = 'server:driver:location';
  static const String serverDriverArrived = 'server:driver:arrived';
  static const String serverRideStarted = 'server:ride:started';
  static const String serverRideCompleted = 'server:ride:completed';
  static const String serverDriverCancelled = 'server:driver:cancelled';

  // Admin (driver doesn't receive these; listed for parity)
  static const String adminSosAlert = 'admin:sos:alert';
  static const String adminDashboard = 'admin:dashboard';
}
